File: cpp/tests/PascalsTriangleTest.cpp
#include "PascalsTriangle.h"
#include "gtest/gtest.h"

using namespace lcpp;

TEST(PascalsTriangle, Solution118_1) {
  auto S118_1 = Solution118_1();

  int const NumRows1 = 5;
  std::vector<std::vector<int>> const Expected1{
      {1}, {1, 1}, {1, 2, 1}, {1, 3, 3, 1}, {1, 4, 6, 4, 1}};
  EXPECT_EQ(Expected1, S118_1.generate(NumRows1));

  int const NumRows2 = 1;
  std::vector<std::vector<int>> const Expected2{{1}};
  EXPECT_EQ(Expected2, S118_1.generate(NumRows2));
}

File: java/src/test/java/io/binac/leetcode/PascalsTriangleTest.java
package io.binac.leetcode;

import org.junit.jupiter.api.Test;

import java.util.Arrays;
import java.util.Collections;
import java.util.List;

import static org.junit.jupiter.api.Assertions.*;

class PascalsTriangleTest {
    private static final PascalsTriangle.Solution1 SOLUTION1 = new PascalsTriangle.Solution1();

    private final int NUMROWS1 = 5;
    private final List<List<Integer>> EXPECTED1 = Arrays.asList(
            Collections.singletonList(1),
            Arrays.asList(1, 1),
            Arrays.asList(1, 2, 1),
            Arrays.asList(1, 3, 3, 1),
            Arrays.asList(1, 4, 6, 4, 1)
    );

    private final int NUMROWS2 = 1;
    private final List<List<Integer>> EXPECTED2 = Collections.singletonList(
            Collections.singletonList(1)
    );

    @Test
    void test1() {
        assertEquals(EXPECTED1, SOLUTION1.generate(NUMROWS1));
        assertEquals(EXPECTED2, SOLUTION1.generate(NUMROWS2));
    }
}